src/agent/debuglet.ts

```diff
@@ -114,6 +120,9 @@ export class Debuglet extends EventEmitter {
   private controller_: Controller;
   private completedBreakpointMap_: {[key: string]: boolean};
 
+  private promise: Promise<void>;
```

Comment:
As another point of view, you could also make a `IsReadyManager` (or something similar) class that contains these few properties and exposes `isReady` and `resolvePromise` methods (or use some better names :)).  That way those methods could be documented and tested in isolation and the `Debuglet` would just use an instance of `IsReadyManager`.  This would allow for separation of concerns.  Its just a different way of looking at the problem.

Reply:
+1; at the very least `promise*` should be changed to something more descriptive.